jdaviz/app.py

```diff
@@ -759,8 +759,22 @@ def _link_new_data(self, reference_data=None, data_to_be_linked=None):
                  linked_data.ndim < 3 and  # Cube linking requires special logic. See below
                  ref_data.ndim < 3)
               ):
-            links = [LinkSame(linked_data.components[0], ref_data.components[0]),
-                     LinkSame(linked_data.components[1], ref_data.components[1])]
+            if self.config == 'specviz2d':
```

Comment:
Hmm does this render #3434 unnecessary? Did you see if the modeling bug went away only with this patch?